pkg/probe/http/http_test.go

```diff
@@ -165,6 +165,55 @@ func TestHTTPProbeChecker(t *testing.T) {
 			health:     probe.Success,
 			accBody:    "User-Agent: kube-probe/",
 		},
+		{
+			handler: headerEchoHandler,
+			reqHeaders: http.Header{
+				"User-Agent": {"foo/1.0"},
```

Comment:
does this test case has significance comparing to one without User-Agent below?

Reply:
The idea is that since the `headers` object is being updated when there is a custom `Accept` and when there is custom `User-Agent`  I wanted to make sure one didn't ruin the other when both are customized. It could be that for some reason setting a custom `Accept` would delete/change the `User-Agent` and vice-versa. 

Another option would (/can) be to improve the assert function to allow assertion of 2 headers in same test but thought it would be easier like this.

Reply:
perhaps then add another test with non-specific header `Foo` and `Accept` together. Not critical

Reply:
got it, will look into both suggestions. thanks!

Reply:
Ok, I added tests to verify the default headers and a test with a custom header + Accept

Please take a look

Reply:
thank you!